File: cms/src/main/java/com/tqp/cms/service/NotificationService.java
package com.tqp.cms.service;

import com.tqp.cms.dto.request.NotificationCreationRequest;
import com.tqp.cms.dto.request.NotificationUpdateRequest;
import com.tqp.cms.dto.response.NotificationListResponse;
import com.tqp.cms.dto.response.NotificationResponse;
import java.util.UUID;
import org.springframework.data.domain.Page;

public interface NotificationService {
    NotificationResponse createNotification(NotificationCreationRequest request);

    Page<NotificationListResponse> getNotifications(int page, int size, String title);

    NotificationResponse getNotificationById(UUID notificationId);

    NotificationResponse updateNotification(UUID notificationId, NotificationUpdateRequest request);

    void softDeleteNotification(UUID notificationId);

    Page<NotificationListResponse> getMyNotifications(int page, int size);
}

File: clinic-app/src/services/notificationService.js
import { api } from './api.js'

export async function getNotifications(params) {
  const { data } = await api.get('/notifications', { params })
  return data
}

export async function getNotificationById(notificationId) {
  const { data } = await api.get(`/notifications/${notificationId}`)
  return data
}

export async function createNotification(payload) {
  const { data } = await api.post('/notifications', payload)
  return data
}

export async function updateNotification(notificationId, payload) {
  const { data } = await api.patch(`/notifications/${notificationId}`, payload)
  return data
}

export async function deleteNotification(notificationId) {
  await api.delete(`/notifications/${notificationId}`)
}

export async function getMyNotifications(params) {
  const { data } = await api.get('/notifications/me', { params })
  return data
}

